llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -1078,10 +1070,12 @@ void SIWholeQuadMode::lowerBlock(MachineBasicBlock &MBB) {
       ActiveLanesReg = 0;
       break;
     case AMDGPU::V_SET_INACTIVE_B32:
-    case AMDGPU::V_SET_INACTIVE_B64:
       if (ActiveLanesReg) {
-        MI.addOperand(*MBB.getParent(),
-                      MachineOperand::CreateReg(ActiveLanesReg, false, true));
+        LiveInterval &LI = LIS->getInterval(MI.getOperand(5).getReg());
+        MRI->constrainRegClass(
+            ActiveLanesReg, TRI->getRegClass(AMDGPU::SReg_1_XEXECRegClassID));
```

Comment:
SReg_1_* classes shouldn't be needed at this point? Don't those only exist as a DAG hack? Can directly query the correct wave mask class

Reply:
`getRegClass` converts to the appropriate 32 or 64 bit class for me. That's the whole point of calling it here.

Reply:
Calling getWaveMaskRegClass is more straightforward?

Reply:
Thanks, I didn't know about that.